swap_meet/vendor.py

```
@@ -1,2 +1,76 @@
 class Vendor:
-    pass
+    def __init__(self, inventory = None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+        
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+        
+    def get_by_category(self, category):
+        return_list = []
+        for item in self.inventory:
+            if item.category == category:
+                return_list.append(item)
+        return return_list
+    
+    def swap_items(self, friend_vendor, my_item, friend_item):
+        try:
+            if not self.remove(my_item):
+                return False
+
+            if not friend_vendor.remove(friend_item):
+                # If swap cannot be completed, put the removed item back in the self inventory
+                self.add(my_item)
+                return False
+
+            self.add(friend_item)                
+            friend_vendor.add(my_item)
+
+            return True
+        except Exception as e:
+            print(str(e))
+            return False
+
+    def swap_first_item(self, friend_vendor):
+        try:
+            return self.swap_items(friend_vendor, self.inventory[0], friend_vendor.inventory[0])
+        except Exception as e:
```

It's ok to catch all exceptions in this way, but if you're anticipating a particular error, such as an IndexError if either self.inventory or friend_vendor.inventory is empty, I would recommend catching that error explicitly to indicate that it's a predicted situation.